src/WebJobs.Script/StorageProvider/AzureStorageProvider.cs

```diff
@@ -18,16 +20,37 @@ namespace Microsoft.Azure.WebJobs.Script
     internal class AzureStorageProvider : IAzureStorageProvider
     {
         private readonly BlobServiceClientProvider _blobServiceClientProvider;
-        private readonly IConfiguration _configuration;
-        private readonly ILogger _logger;
+        private readonly IServiceProvider _serviceProvider;
         private IOptionsMonitor<JobHostInternalStorageOptions> _storageOptions;
+        private IConfiguration _configuration;
+        private ILogger _logger;
 
-        public AzureStorageProvider(IConfiguration configuration, BlobServiceClientProvider blobServiceClientProvider, IOptionsMonitor<JobHostInternalStorageOptions> options, ILogger<AzureStorageProvider> logger)
+        public AzureStorageProvider(IScriptHostManager scriptHostManager, IConfiguration configuration, BlobServiceClientProvider blobServiceClientProvider, IOptionsMonitor<JobHostInternalStorageOptions> options, ILogger<AzureStorageProvider> logger)
         {
             _blobServiceClientProvider = blobServiceClientProvider;
-            _configuration = configuration;
+            _serviceProvider = scriptHostManager as IServiceProvider;
             _storageOptions = options;
             _logger = logger;
+
+            if (FeatureFlags.IsEnabled(ScriptConstants.FeatureFlagDisableMergedWebHostScriptHostConfiguration))
+            {
+                _configuration = configuration;
+            }
+            else
+            {
+                _configuration = new ConfigurationBuilder()
+                .AddConfiguration(configuration)
+                .Add(new ActiveHostConfigurationSource(_serviceProvider))
+                .Build();
+            }
+        }
+
+        public IConfiguration Configuration
```

Comment:
As noted above, hopefully, after the suggested changes, this property won't be required.